recipe/build-pyarrow.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/sh
+
+set -e
+set -x
+
+# Build dependencies
+export ARROW_HOME=$PREFIX
+export PARQUET_HOME=$PREFIX
+export SETUPTOOLS_SCM_PRETEND_VERSION=$PKG_VERSION
+export PYARROW_BUILD_TYPE=release
+export PYARROW_WITH_DATASET=1
+export PYARROW_WITH_FLIGHT=1
+if [[ "$(uname -m)" = "ppc64le" || "$(uname -m)" = "aarch64" ]]
+then
+  export PYARROW_WITH_GANDIVA=0
+else
+  export PYARROW_WITH_GANDIVA=1
+fi
+export PYARROW_WITH_HDFS=1
+export PYARROW_WITH_ORC=1
+export PYARROW_WITH_PARQUET=1
+export PYARROW_WITH_PLASMA=1
+export PYARROW_WITH_S3=1
+BUILD_EXT_FLAGS=""
+
+# Enable CUDA support
+if [[ ! -z "${cuda_compiler_version+x}" && "${cuda_compiler_version}" != "None" ]]
+then
+    export PYARROW_WITH_CUDA=1
+    BUILD_EXT_FLAGS="${BUILD_EXT_FLAGS} --with-cuda"
+else
+    export PYARROW_WITH_CUDA=0
+fi
+
+cd python
+
+$PYTHON setup.py \
+        build_ext $BUILD_EXT_FLAGS \
+        install --single-version-externally-managed \
+                --record=record.txt
+
+# Test CUDA support
+if [[ "$PYARROW_WITH_CUDA" = "1" ]]
+then
+    # move out from pyarrow source directory
+    mkdir tmp-test
+    cd tmp-test
+    wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/cuda-compat-10-0_410.48-1_amd64.deb
+    ar x cuda-compat-10-0_410.48-1_amd64.deb
```

Comment:
Aren't these files in `/usr/local/cuda-10.0/compat/`?

Reply:
I mean already in the docker image?

Reply:
No, not when running build-locally.py. Let me recheck..

Reply:
```
+ ls -la /usr/local/
total 52
drwxr-xr-x 1 root root 4096 May 30  2019 .
drwxr-xr-x 1 root root 4096 Oct  6  2018 ..
drwxr-xr-x 2 root root 4096 Sep 23  2011 bin
lrwxrwxrwx 1 root root    8 May 30  2019 cuda -> cuda-9.2
drwxr-xr-x 1 root root 4096 May 30  2019 cuda-9.2
drwxr-xr-x 2 root root 4096 Sep 23  2011 etc
drwxr-xr-x 2 root root 4096 Sep 23  2011 games
drwxr-xr-x 2 root root 4096 Sep 23  2011 include
drwxr-xr-x 2 root root 4096 Sep 23  2011 lib
drwxr-xr-x 2 root root 4096 Sep 23  2011 lib64
drwxr-xr-x 2 root root 4096 Sep 23  2011 libexec
drwxr-xr-x 2 root root 4096 Sep 23  2011 sbin
drwxr-xr-x 5 root root 4096 Oct  6  2018 share
drwxr-xr-x 2 root root 4096 Sep 23  2011 src
```
The `compat` was introduced for cuda-10. Should we switch from 9.2 to 10.0?

Reply:
Although, switching to 10.0 does not sound good as arrow-cpp is built using 9.2..
Btw, `/usr/local/cuda-9.2/compat/` does not exist either.

Reply:
10.0 was the first version to be introduced with a driver compatibility mode if I remember correct.

Reply:
> Although, switching to 10.0 does not sound good as arrow-cpp is built using 9.2.

While building for 9.2 should not be skipped, you could *also* build for `10.x` (even if not part of the CI upstream), as long as the build completes and some reasonable choice of tests passes.

Reply:
@pearu have you thought about this suggestion? How about building for CUDA 10.x as well (at least trying once)?

Reply:
So, the question is whether the cuda 10.0 docker image will have cuda-compat available or not.
We could check that on CI by enabling 10.0 as a temporary patch.

Reply:
Yes, 10.x all have cuda-compat. Otherwise I wouldn't do https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/93 just for 9.2, but for all where necessary.

Reply:
Proof:
```
xxx@yyy:~$ sudo docker run --rm -it nvidia/cuda:10.0-devel-centos6 bash
[root@4e5ebf4565bb /]# find /usr/ -name 'libcuda.so*'
/usr/local/cuda-10.0/targets/x86_64-linux/lib/stubs/libcuda.so
/usr/local/cuda-10.0/compat/libcuda.so.410.129
/usr/local/cuda-10.0/compat/libcuda.so
/usr/local/cuda-10.0/compat/libcuda.so.1
```